src/native/corehost/hostmisc/pal.windows.cpp

```diff
@@ -898,8 +898,12 @@ bool pal::realpath(pal::string_t* path, bool skip_error_logging)
                 }
             }
 
-            // Remove the \\?\ prefix, unless it is necessary or was already there
-            if (LongFile::IsExtended(str) && !LongFile::IsExtended(*path) &&
+            // Remove the UNC prefix (\\?\UNC\) or extended prefix (\\?\) unless it is necessary or was already there
+            if (LongFile::IsUNCExtended(str) && !LongFile::IsUNCExtended(*path) && !LongFile::ShouldNormalize(*path))
```

Comment:
Wouldn't that always be false? That is, the resolved string that comes out should always be normalized. My intent here was that, if the path that came in was non-normalized, then we should normalize it. But if it was "absolute" before, and it's a network path (because `\\?\UNC\` is always a network path, if I'm reading this right) then we can stick with the original path.

I guess the question is specifically for network paths -- are we OK with keeping the original path? Or do we want to try the UNC extended path back into a "conventional" UNC path?

Reply:
Can symlinks exist for network paths? I didn’t think so. But I should look into that more.

Reply:
Looks like they can, will fix.